lib/iris/experimental/ugrid/load.py

```diff
@@ -485,9 +485,27 @@ def _build_mesh_coords(mesh, cf_var):
         "edge": mesh.edge_dimension,
         "face": mesh.face_dimension,
     }
-    mesh_dim_name = element_dimensions[cf_var.location]
-    # (Only expecting 1 mesh dimension per cf_var).
-    mesh_dim = cf_var.dimensions.index(mesh_dim_name)
+    location = getattr(cf_var, "location", "<empty>")
```

Comment:
N.B. This relates to https://github.com/SciTools/iris/issues/5165

It would be really nice to be more "tolerant" here : I had hoped to issue appropriate warnings (instead of errors), and simply not attach the mesh in any of these problematic cases.

However, when I tried to code that, I hit problems with **_testing_** it, mostly because of the context in which the UGRID code was written : 
  * it is using logging instead of warnings, which was probably is mis-step + wants "changing back" (and is awkward to test for)
  * the UGRID loading code has a _particularly_ "hardline" approach to the unexpected : mostly just "fail if not as expected" -- e.g. in [the motivating issue](https://github.com/SciTools/iris/issues/5908), you just get an obscure indexing error.  So, my attempts to code some "tolerant loading in this respect only" just hit a bunch of snags, adding a lot of additional complexity in unrelated areas

**_So,_** to avoid exploding the cost beyond the scope of this current effort, I instead cut my losses here + settled for just producing more intelligible error messages.
Hopefully we can still address the tolerant loading approach in future, but right now that seems to be a bigger challenge for the UGRID code than the 'regular' netcdf loading